components/buy/Buy.vue

```diff
@@ -236,6 +239,16 @@ const watchNftsPriceChanges = (nftIds: string[], force: boolean = false) => {
   })
 }
 
+watch(
+  () => preferencesStore.completePurchaseModal.isOpen,
+  (isOpen, prevIsOpen) => {
+    const modalGotClosed = !isOpen && prevIsOpen
+    if (modalGotClosed) {
+      nftSubscription.unsubscribe()
+    }
+  },
+)
+
 watch(
   [
     () => preferencesStore.completePurchaseModal.isOpen,
```

Comment:
can we add this the other watcher , we already listen for `preferencesStore.completePurchaseModal.isOpen` there

Reply:
idk, that watcher watches 3 things so it's hard to reason about
i want simple:
modal open => subscribe
modal close => unsubscribe

Reply:
fair enough